### PR TITLE
Lcam 402 cru correspondence state

### PR DESCRIPTION
**Pull request by AafAnsari**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-402)

Added controller, service, repository, entity and test classes for `correspondence_state` table

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/test/java/gov/uk/courtdata/correspondence/controller/CorrespondenceStateControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(CorrespondenceStateController.class)
+public class CorrespondenceStateControllerTest {
+
+    private static final String ENDPOINT_URL = "/api/internal/v1/assessment/correspondence-state";
+    private static int INVALID_REP_ID = 1235;
```
Comment (stone-dj) [inline maat-court-data-api/src/test/java/gov/uk/courtdata/correspondence/controller/CorrespondenceStateControllerTest.java:28]: `INVALID_REP_ID` should also be declared `final`

File: maat-court-data-api/src/test/java/gov/uk/courtdata/correspondence/service/CorrespondenceStateServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class CorrespondenceStateServiceTest {
+
+    private static int INVALID_REP_ID = 1235;
```
Comment (stone-dj) [inline maat-court-data-api/src/test/java/gov/uk/courtdata/correspondence/service/CorrespondenceStateServiceTest.java:23]: `INVALID_REP_ID` should be also declared `final`

**Review by stone-dj — CHANGES_REQUESTED** [review]
Changes look good, I like the thorough and easy to read unit tests

File: maat-court-data-api/src/test/java/gov/uk/courtdata/correspondence/controller/CorrespondenceStateControllerTest.java (hunk 1 above)
Comment (AafAnsari) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/correspondence/controller/CorrespondenceStateControllerTest.java:28]: yes I agree, missed it

**Review by stone-dj — APPROVED** [review]
Changes look good, no further feedback from myself

**Pull request merged**